libpod/networking_common.go

```diff
@@ -632,14 +640,22 @@ func (r *Runtime) ConnectContainerToNetwork(nameOrID, netName string, netOpts ty
 	return ctr.NetworkConnect(nameOrID, netName, netOpts)
 }
 
-// normalizeNetworkName takes a network name, a partial or a full network ID and returns the network name.
+// normalizeNetworkName takes a network name, a partial or a full network ID and
+// returns: 1) the network name and 2) the network_interface name for macvlan
+// and ipvlan drivers if the naming pattern is "device". Else, "".
 // If the network is not found an error is returned.
-func (r *Runtime) normalizeNetworkName(nameOrID string) (string, error) {
+func (r *Runtime) normalizeNetworkName(nameOrID, namingPattern string) (string, string, error) {
```

Comment:
No need for the second param, just use `r.config.Containers.InterfaceName` directly here instead of forcing the caller do the right thing.